ext/authx/Civi/Authx/Authenticator.php

```diff
@@ -190,6 +194,9 @@ protected function checkCredential($tgt) {
     \Civi::dispatcher()->dispatch('civi.authx.checkCredential', $checkEvent);
 
     if ($checkEvent->getRejection()) {
+      if ($tgt->redirect) {
+        _authx_reload($tgt->redirect, 'msg="' . $checkEvent->getRejection() . '"');
```

Comment:
I was concerned about this becoming vector for an "open redirect" (https://learn.snyk.io/lesson/open-redirect/). But I think (through a bit of happenstance) it turns out OK as-is. Specifically, if you trace `_authx_reload()`, the `$redirect` is passed through `CRM_Utils_System::url()`:

* If `url()` runs in the default mode (`$absolute==FALSE`), then you could give a malicious value (`/example.com/foobar`) which expands to a legal/external URL (`//example.com/foobar`  -- ie `example.com` with an inherited scheme, `http:` or `https:`).
    * (_This may depend on the usage of clean URLs. My local dmaster behaved this way, but my local wpmaster didn't._)
* In actuality, `url()` is called with `$absolute==TRUE`. In this mode, I don't see a way to trick it into an external redirect.
    * (_I don't remember any particular reason for this flag -- just a feeling that absolute URLs are vaguely more reliable._)
    * (_This is basically what `snyk.io` refers to as "Implement a fixed domain"._)

I can imagine someone innocently flipping that to insecure mode... so I pushed up a small comment to explain that it's now important.